packages/playwright/src/takeSnapshot.ts

```diff
@@ -35,7 +35,15 @@ async function takeSnapshot(
   // Serialize and capture the DOM
   const domSnapshot: elementNode = await page.evaluate(dedent`
     ${rrweb};
-    rrwebSnapshot.snapshot(document);
+    if (typeof define === "function" && define.amd) {
+      new Promise((resolve) => {
```

Comment:
One small thing -- is there a reason we're wrapping this in a `Promise` here? Since we're not awaiting the results of the promise.

Reply:
I don't quite understand why, but this doesn't work if I remove the Promise

Reply:
Maybe playwright is waiting on the promise somehow? Maybe we should `return new Promise...` just to make that 100% clear.

Reply:
I don't see how `if (x) { y } else { z }` evaluates to anything, but clearly it does somehow (I assume we use the return value `domSnapshot` still right?).

Maybe there is some way to wrap this in a function (so you can return) to make it less confusing?

Reply:
I added additional comments to hopefully point out the odd behavior that `page.evaluate` displays, along with a link to Playwright doc. I'd like to avoid using a ternary since it's possible we might have other scenarios to account for in the future.